verde/model_selection.py

```diff
@@ -7,17 +8,272 @@
 import warnings
 
 import numpy as np
-from sklearn.model_selection import KFold, ShuffleSplit
+from sklearn.model_selection import KFold, ShuffleSplit, BaseCrossValidator
 from sklearn.base import clone
 
-from .base import check_fit_input
+from .base import check_fit_input, n_1d_arrays
+from .coordinates import block_split
 from .utils import dispatch
 
 
 # Otherwise, DeprecationWarning won't be shown, kind of defeating the purpose.
 warnings.simplefilter("default")
 
 
+class BlockShuffleSplit(BaseCrossValidator):
+    """
+    Random permutation of spatial blocks cross-validator.
```

Comment:
I'm not very happy with this but I wanted to stay close to the sklearn description. Any suggestions?

Reply:
At first read it seems like a strange description, but I think it succinctly describes the class. The other ideas I had were just different versions saying the same thing but in more words. I'll think about this for a while